### PR TITLE
nilrt_ip: Fix DNS

### PR DESCRIPTION
**Pull request by heghe**
Configuring DNS in connman is done by using Nameservers section not
Domain section

Signed-off-by: Heghedus Razvan <razvan.heghedus@ni.com>

### What does this PR do?
Fix an incorrect behaviour where instead of setting the dns was set the domain

### Previous Behavior
set_static_all was setting the domain(wrong behaviour)

### New Behavior
set_static_all is setting the nameservers(aka dns)

### CODE REVIEW

File: salt/modules/nilrt_ip.py [hunk 1]
```diff
@@ -351,39 +351,46 @@ def set_dhcp_linklocal_all(interface):
...
-    validate, msg = _space_delimited_list(domains)
-    if not validate:
-        raise salt.exceptions.CommandExecutionError(msg)
+    if nameservers is not '':
+        validate, msg = _space_delimited_list(nameservers)
+        if not validate:
+            raise salt.exceptions.CommandExecutionError(msg)
+        if not isinstance(nameservers, list):
+            dns = nameservers.split(' ')
+            nameservers = dns
```
Comment (skizunov) [inline salt/modules/nilrt_ip.py]: Could this be simplified to `nameservers = nameservers.split(' ')` or does that trigger a pylint error?

File: salt/modules/nilrt_ip.py [hunk 2]
```diff
@@ -392,10 +399,8 @@ def set_static_all(interface, address, netmask, gateway, domains):
     ipv4['Gateway'] = dbus.String('{0}'.format(gateway), variant_level=1)
     try:
         service.set_property('IPv4.Configuration', ipv4)
-        if not isinstance(domains, list):
-            dns = domains.split(' ')
-            domains = dns
-        service.set_property('Domains.Configuration', [dbus.String('{0}'.format(d)) for d in domains])
+        if nameservers is not '':
```
Comment (skizunov) [inline salt/modules/nilrt_ip.py]: Perhaps just (to be more pythonic):

```
if nameservers:
```

File: salt/modules/nilrt_ip.py [hunk 3]
```diff
@@ -351,39 +351,46 @@ def set_dhcp_linklocal_all(interface):
...
     service = _interface_to_service(interface)
     if not service:
         raise salt.exceptions.CommandExecutionError('Invalid interface name: {0}'.format(interface))
     validate, msg = _validate_ipv4([address, netmask, gateway])
     if not validate:
         raise salt.exceptions.CommandExecutionError(msg)
-    validate, msg = _space_delimited_list(domains)
-    if not validate:
-        raise salt.exceptions.CommandExecutionError(msg)
+    if nameservers is not '':
```
Comment (skizunov) [inline salt/modules/nilrt_ip.py]: Perhaps just (to be more pythonic):

```
if nameservers:
```

**Review by skizunov — CHANGES_REQUESTED** [review]

File: salt/modules/nilrt_ip.py (hunk 3 above)
Comment (raazvvann) [reply salt/modules/nilrt_ip.py]: I just want to be sure. I will change it. Same below

File: salt/modules/nilrt_ip.py (hunk 1 above)
Comment (raazvvann) [reply salt/modules/nilrt_ip.py]: I donn't know. I'll try and see if it will complain.

**Comment by cachedout** [comment]
@skizunov Could you please re-review these changes and let me know if they look good to you? Thanks.

**Review by skizunov — APPROVED** [review]

**Comment by skizunov** [comment]
@cachedout, @raazvvann : The changes look good to me.

**Review by cachedout — APPROVED** [review]

**Pull request merged**